tests/neg-macros/i9801/Test_2.scala

```diff
@@ -0,0 +1 @@
+def test: Unit = loop(4) // error
```

Comment:
Now, this reports correctly the NPE
```scala
-- Error: tests/pos-macros/i9801/Test_2.scala:1:21 ---------------------------------------------------------------------
1 |def test: Unit = loop(4)
  |                 ^^^^^^^
  |       Exception occurred while executing macro expansion.
  |       java.lang.NullPointerException
  |             at dotty.tools.dotc.reporting.Message.dropNonSensical(Message.scala:65)
  |             at dotty.tools.dotc.reporting.Message.message(Message.scala:78)
  |             at dotty.tools.dotc.reporting.Message.isNonSensical(Message.scala:90)
  |             at dotty.tools.dotc.reporting.HideNonSensicalMessages.isHidden(HideNonSensicalMessages.scala:16)
  |             at dotty.tools.dotc.reporting.HideNonSensicalMessages.isHidden$(HideNonSensicalMessages.scala:10)
  |             at dotty.tools.dotc.reporting.TestReporter.isHidden(TestReporter.scala:18)
  |             at dotty.tools.dotc.reporting.Reporter.report(Reporter.scala:148)
  |             at dotty.tools.dotc.report$.error(report.scala:71)
  |             at dotty.tools.dotc.quoted.reflect.ReflectionCompilerInterface.error(ReflectionCompilerInterface.scala:90)
  |             at dotty.tools.dotc.quoted.QuoteContextImpl$$anon$1.error(QuoteContextImpl.scala:42)
  |             at Macro_1$package$.impl(Macro_1.scala:18)
  |
  | This location contains code that was inlined from Test_2.scala:2

inlined at Test_2.scala:1:
```

Reply:
Notice that in this particular case the user passed a `null` into the reporting which made the reporting itself fail.

Reply:
You mean `e.getMessage` returns `null`?